src/main/java/de/tum/in/www1/artemis/web/rest/metis/PostResource.java

```diff
@@ -63,6 +63,38 @@ public PostResource(PostService postService) {
         return new ResponseEntity<>(updatedPost, null, HttpStatus.OK);
     }
 
+    /**
+     * PUT /courses/{courseId}/posts/{postId}/pin : Pin an existing post by setting the pinned flag to true (is used as criterion for sorting the posts)
+     *
+     * @param courseId  id of the course the post belongs to
+     * @param postId    id of the post to pin
+     * @param pinState  updated boolean value of the pinned flag for the given post
+     * @return ResponseEntity with status 200 (OK) containing the updated post in the response body,
+     * or with status 400 (Bad Request) if the checks on user, course or post validity fail
+     */
+    @PutMapping("courses/{courseId}/posts/{postId}/pin")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<Post> updatePinState(@PathVariable Long courseId, @PathVariable Long postId, @RequestBody Boolean pinState) {
+        Post postWithUpdatedPinState = postService.updatePinState(courseId, postId, pinState);
+        return ResponseEntity.ok().body(postWithUpdatedPinState);
+    }
+
+    /**
+     * PUT /courses/{courseId}/posts/{postId}/archive : Archive an existing post by setting the archive flag to true (is used as criterion for sorting the posts)
+     *
+     * @param courseId      id of the course the post belongs to
+     * @param postId        id of the post to vote on
+     * @param archiveState  updated boolean value of the archived flag for the given post
+     * @return ResponseEntity with status 200 (OK) containing the updated post in the response body,
+     * or with status 400 (Bad Request) if the checks on user, course or post validity fail
+     */
+    @PutMapping("courses/{courseId}/posts/{postId}/archive")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<Post> updateArchiveState(@PathVariable Long courseId, @PathVariable Long postId, @RequestBody Boolean archiveState) {
```

Comment:
Should this be wrapped in something like
```
if (authorizationCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
```
? I think right now every user with `TA` permissions in any course can call this.

Reply:
see above